edx_proctoring/views.py

```diff
@@ -1104,3 +1107,40 @@ def post(self, request, user_id):  # pylint: disable=unused-argument
                         code = 500
                 seen.add(backend_name)
         return Response(data=results, status=code)
+
+
+class UserRetirement(AuthenticatedAPIView):
+    """
+    Retire user personally-identifiable information (PII) for a user
+    """
+    def _retire_exam_attempts_user_info(self, user_id):
+        """ Obfuscate PII for exam attempts and exam history """
+        attempts = ProctoredExamStudentAttempt.objects.filter(user_id=user_id)
+        if attempts:
+            for attempt in attempts:
+                attempt.student_name = obscured_user_id(attempt.student_name)
+                attempt.last_poll_ipaddr = obscured_user_id(attempt.last_poll_ipaddr)
```

Comment:
This seems like a poorly named function if we're using it for non-id values. I'm not familiar enough with proctoring to know if it's an option, but edx-platform has special methods for hashing some values (username, email at least) that take advantage of the ability to rotate a configurable seed instead of relying on the Django secret key, which is a bigger deal to rotate. We've actually had to do this a couple of times. If you can use those, I'd suggest it.

Reply:
@bmedx, this is outdated since we're now just clearing values. Was reusing as a convenient hashing function but I agree, maybe not the best way to have done it 👍